docsbox/docs/tasks.py

```diff
@@ -41,27 +42,33 @@ def create_tmp_file_and_get_mimetype(original_file, filename, stream=False, dele
 
 @rq.job(timeout=app.config["REDIS_JOB_TIMEOUT"])
 def process_convertion(path, options, meta):
-    current_task = get_current_job()
-    exportFormatType = app.config["CONVERTABLE_MIMETYPES"][meta["mimetype"]]["formats"]
-    if exportFormatType in app.config["DOCUMENT_CONVERTION_FORMATS"]:
-        result= process_document_convertion(path, options, meta, current_task)
-    elif exportFormatType == "IMAGE_EXPORT_FORMATS":
-        result= process_image_convertion(path, options, meta, current_task)
-    elif exportFormatType == "AUDIO_EXPORT_FORMATS":
-        result= process_audio_convertion(path, options, meta, current_task)
-    elif exportFormatType == "VIDEO_EXPORT_FORMATS":
-        result= process_video_convertion(path, options, meta, current_task)
-
-    if result and meta["via_allowed_users"]:
-        r = save_file_on_via(app.config["MEDIA_PATH"] + current_task.id, result["mimeType"], meta["via_allowed_users"])
-        remove_file(app.config["MEDIA_PATH"] + current_task.id)
-        result['fileId'] = r.headers.get("Document-id")
-    return result
+    try:
+        current_task = get_current_job()
+        exportFormatType = app.config["CONVERTABLE_MIMETYPES"][meta["mimetype"]]["formats"]
+        if exportFormatType in app.config["DOCUMENT_CONVERTION_FORMATS"]:
+            result= process_document_convertion(path, options, meta, current_task)
+        elif exportFormatType == "IMAGE_EXPORT_FORMATS":
+            result= process_image_convertion(path, options, meta, current_task)
+        elif exportFormatType == "AUDIO_EXPORT_FORMATS":
+            result= process_audio_convertion(path, options, meta, current_task)
+        elif exportFormatType == "VIDEO_EXPORT_FORMATS":
+            result= process_video_convertion(path, options, meta, current_task)
+
+        if result and meta["via_allowed_users"]:
+            r = save_file_on_via(app.config["MEDIA_PATH"] + current_task.id, result["mimeType"], meta["via_allowed_users"])
+            remove_file(app.config["MEDIA_PATH"] + current_task.id)
+            result['fileId'] = r.headers.get("Document-id")
+        return result
+    except Exception as e:
+        return { "has_failed": True, "message": e , "traceback": traceback.format_exc() }
 
 def process_document_convertion(path, options, meta, current_task):
     output_path = os.path.join(app.config["MEDIA_PATH"], current_task.id)
     if (meta["mimetype"] == "application/pdf"):
-        os.system("ocrmypdf --tesseract-timeout=0 --optimize 0 --skip-text {0} {1}".format(path, output_path))
+        # faster conversion and file size smaller, if XMP is not found a more forcefull conversion is made so file ends as a PDF/A
+        os.system("ocrmypdf --tesseract-timeout=0 --optimize 0 --redo-ocr --skip-big 30 {0} {1}".format(path, output_path))
```

Comment:
What's the redo-ocr param for? 30 megapixel limit for images it still quite high, don't you think?

Reply:
yeah it should be --skip-text I was testing with redo to see if it was better but there is no change and it takes a bit more.
At what size should it skip the image? this is mainly for performance and I think only needed in the --force-ocr.

Reply:
Ok. If it's only for the ones which have no XMP, we should make it so also then I think we could keep the 30 since it's not so common